### ISSUE TITLE
Keep the link when building wheels.

### ISSUE THREAD

**Issue by wqh17101** [pypa/wheel#453]
Hi there,
I have a dir to pack. And there are some links in it.Like
```
a.so
a.so.1 -> a.so
b.so
b.so.1 -> b.so
```
but when i build wheels, it generated a package that
```
a.so
a.so.1 
b.so
b.so.1
```
It lost their links. It cause that the size of package is double than expected.

**Comment by wqh17101** [comment]
Maybe https://github.com/pypa/wheel/blob/c05b6b409defef5b59449d230683dbf89633fad0/src/wheel/wheelfile.py#L142 cause the problem.

**Comment by pfmoore** [comment]
There's an existing discussion [here](https://discuss.python.org/t/symbolic-links-in-wheels/1945). Better to contribute there than to open a new topic and split the conversation.

**Comment by leofang** [comment]
xref: https://github.com/pypa/wheel/issues/203

**Comment by rgommers** [comment]
Here is another use case besides shipping shared libraries: support for editable installs when the build system uses out-of-place builds. Out-of-place builds are the only option in Meson, and also good practice for CMake. For out-of-place builds, you ends up with compiled extension modules and generated files in the build directory, and `.py` files in the source directory. To put those together into a working editable install, the most straightforward solution is putting symlinks to all files in a wheel and installing those. xref https://github.com/mesonbuild/meson-python/issues/47 for more details.

**Comment by agronholm** [comment]
There's no point in asking for this here, when the wheel spec itself doesn't even support symlinks. If that is ever added to the spec, a new issue should be created.

**Comment by rgommers** [comment]
@agronholm I just provided a second use case, which seemed to be missing so far. Guess I'll take it over to the Discourse thread.

You may want to close gh-400 as well.

**Comment by pfmoore** [comment]
@rgommers Symlink support in wheels seems to me like a very useful thing to have. But as @agronholm says, it's not currently part of the wheel spec, so it would need a PEP and a new wheel version. You might be interested in [this post](https://discuss.python.org/t/require-packing-wheels-without-files-in-the-root-directory/21599/19) I just made, and its associated thread, as well as the thread I linked to previously.
